### ISSUE TITLE
Unable to create Object with Object.new()

### ISSUE THREAD

**Issue by Rubonnek** [godotengine/godot#44444]
**Godot version:**

3.2 branch at commit fd6b3060c1cca10754c47ad04a4e50d49ca838b7

**OS/device including version:**

Arch Linux

**Issue description:**

The [`Object` documentation](https://docs.godotengine.org/en/stable/classes/class_object.html#description) states that it's possible to construct `Object` with `Object.new()`, but this is not the case.

**Steps to reproduce:**
Try calling `Object.new()` on a script. You'll get the following:

```
SCRIPT ERROR: GDScript::reload: Parse Error: Static constant  'new' not present in built-in type Object.
```

**Minimal reproduction project:**

```gdscript
#!/usr/bin/godot -s
extends SceneTree

func _init():
	var test = Object.new()
	test.free()
	quit()
```

**Comment by Rubonnek** [comment]
I'm not sure if this issue should be considered a bug or not since `new()` works for other non built-in types, but I find it strange that it does not work for the base class itself.

**Comment by Xrayez** [comment]
This was fixed in 4.0 via #41516 (I cannot confirm myself), but still present in 3.2.

**Comment by Rubonnek** [comment]
Hehe, I was glad with the issue number myself as soon as I saw it.

Wow, last night I searched for `Object.new` and didn't see any issues related to this for some reason. Thanks for the heads up.

This issue is a duplicate of #41462 then, and given that the use cases for plain `Object`s seems quite small imho, I'm closing this issue.

**Comment by Xrayez** [comment]
I'd personally keep this issue open until Godot 4.0 stable is out at least, so that people can find the same issue without creating duplicate reports.

**Comment by Rubonnek** [comment]
Fair enough.

**Comment by akien-mga** [comment]
IMO unless we plan to fix it, this could be mentioned in the 3.2 documentation, and then this issue closed as fixed in 4.0.

**Comment by akien-mga** [comment]
Fixed in 4.0, limitation documented for 3.2 in #44476.
